core/src/packet.rs

```diff
@@ -44,6 +45,30 @@ pub struct Packet {
     pub meta: Meta,
 }
 
+/// Anything we receive over the network
+pub trait Message: Sized + Default + Clone {
```

Comment:
Can you think of an alternative to Message here? If we go with this, then we'd have a Message containing a Transaction containing a Message.